app/routes/docs.$lang.$ref.tsx

```diff
@@ -811,9 +819,12 @@ function MenuCategoryLink({
 }
 
 function MenuLink({ to, children }: { to: string; children: React.ReactNode }) {
+  // Only discover expanded links to keep manifest calls to a reasonable URL length
+  let { isOpen } = React.useContext(MenuCategoryContext);
   let isActive = useIsActivePath(to);
   return (
     <Link
+      discover={isOpen ? "render" : "none"}
```

Comment:
Nice, that's really cool how flexible this API ends up becoming

Only "downside" I can think of is you have more urls to be cached now, but honestly that's not really a big downside, especially since the browser and cdn caches will get filled pretty quickly as the user navigates. Plus, it's smaller urls, like you mentioned, getting loaded incrementally

Man, this is such a cool little heuristic!